pennylane/ftqc/graph_state_preparation.py

```diff
@@ -0,0 +1,89 @@
+    The initial graph is :math:`|0\rangle^{V}`, given each qubit in the graph is in the :math:`|0\rangle` state and is not entangled with each other.
+    The target graph state :math:`| \psi \rangle = \prod\limits_{\{a, b\} \in E} U_{ab}|+\rangle^{V}` is prepared as below:
+    1. Each qubit is prepared as :math:`|+\rangle^{V}` state by applying the `qubit_ops` (`Hadamard` gate) operation.
+    2. Entangle every nearest qubit pair in the graph with `entanglement_ops` (`CZ` gate) operation.
+
+    Args:
+        qubit_graph (QubitGraph): QubitGraph object mapping qubit to wires.
+        qubit_ops (Operations): Operator to prepare the initial state of each qubit. Default as ``qml.H``.
+        entanglement_ops (Operations): Operator to entangle nearest qubits. Default as ``qml.CZ``.
+    """
+
+    def __init__(
+        self,
+        qubit_graph: QubitGraph,
+        qubit_ops: Operation = qml.H,
+        entanglement_ops: Operation = qml.CZ,
+    ):
+        self.hyperparameters["qubit_graph"] = qubit_graph
+        self.hyperparameters["qubit_ops"] = qubit_ops
+        self.hyperparameters["entanglement_ops"] = entanglement_ops
+        super().__init__(
+            None, wires=qubit_graph
+        )  # TODO: check if we have to store the qubit states with this operation object
+
+    def decomposition(self) -> list["Operator"]:
+        r"""Representation of the operator as a product of other operators.
+
+        Returns:
+            list[Operator]: decomposition of the operator
+        """
+        return self.compute_decomposition(**self.hyperparameters)
+
+    @staticmethod
+    def compute_decomposition(
+        qubit_graph: QubitGraph,
+        qubit_ops: Operation = qml.H,
+        entanglement_ops: Operation = qml.CZ,
+    ):
+        r"""Representation of the operator as a product of other operators (static method).
+        .. note::
+
+            Operations making up the decomposition should be queued within the
+            ``compute_decomposition`` method.
+
+        .. seealso:: :meth:`~.Operator.decomposition`.
+
+        Args:
+            qubit_graph (QubitGraph): QubitGraph object mapping qubit to wires.
+            qubit_ops (Operations): Operator to prepare the initial state of each qubit. Default as ``qml.H``.
+            entanglement_ops (Operations): Operator to entangle nearest qubits. Default as ``qml.CZ``.
+
+        Returns:
+            list[Operator]: decomposition of the operator
+        """
+
+        op_list = []
+        # Add qubit_ops for each qubit in the graph
+        for qubit in qubit_graph.graph:
+            op_list.append(qubit_ops(wires=qubit_graph[qubit]))
+
+        # Add entanglement_ops for each pair of nearest qubits in the graph
+        for qubit0, qubit1 in qubit_graph.edges:
+            op_list.append(entanglement_ops(wires=[qubit_graph[qubit0], qubit_graph[qubit1]]))
+        return op_list
```

Comment:
The inherited `map_wires` method doesn't work, but we should wait to resolve that until we decide on exactly how to handle wires